crates/hotshot/src/lib.rs

```diff
@@ -620,10 +632,13 @@ impl<TYPES: NodeType> HotShotInitializer<TYPES> {
     /// initialize from genesis
     /// # Errors
     /// If we are unable to apply the genesis block to the default state
-    pub fn from_genesis(instance_state: TYPES::InstanceState) -> Result<Self, HotShotError<TYPES>> {
+    pub fn from_genesis(
+        instance_state: &TYPES::InstanceState,
```

Comment:
Why switch this back to a reference? We're still consuming it in this function. Just need to use it to construct the validated state *before* consuming it.

Reply:
Yeah, you are right! I should have just constructed the state first.